lib/medtronic/medtronicSimulator.js

```diff
@@ -239,42 +239,46 @@ exports.make = function(config){
             event.suppressed.scheduleName = currPumpSettings.activeSchedule;
           }
           if(currBasal.deliveryType === 'temp') {
-            // nest suppressed scheduled basal in temp basal
+            // nest suppressed scheduled basal in temp basal inside suspended basal
             event.suppressed.suppressed = currBasal.suppressed;
 
-            // also, restart temp basal on resume
-            if(currStatus && currStatus.index === event.index) {
-              // if the indexes are the same, the suspended basal was
-              // created from the same record as the suspend/resume event
-              event.duration = currStatus.duration;
-
-              var resumeBasal = _.clone(currBasal);
-              resumeBasal.time = new Date(Date.parse(currStatus.time) + currStatus.duration).toISOString();
-              resumeBasal.deviceTime = sundial.formatDeviceTime(Date.parse(currStatus.deviceTime) + currStatus.duration);
-              resumeBasal.index = event.resumeIndex;
-              delete resumeBasal.duration; // we don't know the new duration yet
-
-              // finish up the temp basal before the suspend
-              if(currBasal.duration) {
-                currBasal.with_expectedDuration(currBasal.duration);
-                var duration = Date.parse(event.time) - Date.parse(currBasal.time);
-                currBasal.with_duration(duration);
+            if(currBasal.duration > (Date.parse(currStatus.time) - Date.parse(currBasal.time) + currStatus.duration)) {
```

Comment:
Yep, it's the duration from when the temp basal started to when the suspend ended. How about `tempStartToResumeDuration`? Fixed in commit 4dfc03df.